docs/RFCS/20220608_split_jobs_system_tables.md

```diff
@@ -0,0 +1,479 @@
+running job's status to pause-requested. This matches the current behavior with respect to writing a
+pause-request to the current system.jobs, however the goal of eliminating all blocking transactions
+from the new control table suggests that such control statements should be updated to no longer be
+allowed in explicit, user-owned transactions. In many ways, this added restriction should have been
+there already: running `PAUSE JOB X` inside a transaction can be confusing, because it does not
+actually pause the running job until it is committed, at which time a pause-request is written, then
+at some later time observed, and only then does the job itself move to paused. However if adding
+this new restriction proves unpalatable, options include either simply allowing these blocking
+transactions and the fact they block the jobs system --  if a user has indicated they will pause a
+job they need to either commit that fact or abort it before we can answer if the job is paused or
+not -- or we can introduce additional queued operation tables similar to the intake table described
+below. 
+
+
+###### Upgrade Path for system.jobs
+
+Changes to the job control state, as jobs step through the control state machine, would need to
+start writing to both tables, along with job creation (writing to intake). A migration can then copy
+all existing jobs to the new control table in the appropriate state to backfill. Subsequently
+control queries can begin to query the new table instead, until an additional cluster version
+disables writes to the existing table, followed by a migration that drops it. Note that writes to
+job detail and progress are discussed in the job_info section below; this would be for
+control-related writes only.
+
+---
+
+### system.job_status_progress + system.job_status_message
+
+#### Description
+
+The job_status_progress and system.job_status_messages tables store human-readable information about
+a job's current and recent status and progress.
+
+These tables are not read by the jobs control system at any point, and should not be read by the job
+itself either; they are intended solely for human-consumption via SHOW JOBS and the DB Console. 
+
+Both tables have many rows per job: Each time a job updates its persisted status or progress it will
+insert a new row into the respective table with the new value, and inspection tools will scan for
+the latest row per job to find its current status/progress. 
+
+Both tables have the same primary key: job ID plus row creation time. However they are separate as
+their payloads are both of very different size -- a small number versus a large string -- and are
+updated at very different frequencies; a completion percentage or timestamp likely updates
+continuously for the entirely life of a job while execution status may change a fixed number of
+times per execution phase.
+
+#### Schema
+
+```
+    CREATE TABLE system.job_status_progress (
```

Comment:
have you considered making these two tables into one table? I guess we always only need to know the current/latest job progress and status. But I think it's interesting to know at a time point what's the progress and the status. Of course that we can to binary search on both tables, but it'd just be easier with one table. Apologies if I'm making noises.